File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offline/synccenterpayloads/SyncCenterPayloadsMvpView.java
package com.mifos.mifosxdroid.offline.synccenterpayloads;

import com.mifos.mifosxdroid.base.MvpView;
import com.mifos.services.data.CenterPayload;

import java.util.List;

/**
 * Created by mayankjindal on 04/07/17.
 */

public interface SyncCenterPayloadsMvpView extends MvpView {

    void showOfflineModeDialog();

    void showCenters(List<CenterPayload> centerPayloads);

    void showCenterSyncResponse();

    void showCenterSyncFailed(String errorMessage);

    void showPayloadDeletedAndUpdatePayloads(List<CenterPayload> centerPayloads);

    void showCenterPayloadUpdated(CenterPayload centerPayload);

    void showError(int stringId);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offline/synccenterpayloads/SyncCenterPayloadsMvpView.kt
package com.mifos.mifosxdroid.offline.synccenterpayloads

import com.mifos.mifosxdroid.base.MvpView
import com.mifos.services.data.CenterPayload

/**
 * Created by mayankjindal on 04/07/17.
 */
interface SyncCenterPayloadsMvpView : MvpView {
    fun showOfflineModeDialog()
    fun showCenters(centerPayloads: List<CenterPayload>)
    fun showCenterSyncResponse()
    fun showCenterSyncFailed(errorMessage: String)
    fun showPayloadDeletedAndUpdatePayloads(centerPayloads: List<CenterPayload>)
    fun showCenterPayloadUpdated(centerPayload: CenterPayload)
    fun showError(stringId: Int)
}